content/applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/ups_credentials.rst

```diff
@@ -1,68 +1,210 @@
+business`.
+
+In :guilabel:`Choose an account to associate with these credentials`, select :guilabel:`Add existing
+account` and select the :ref:`account number <inventory/shipping_receiving/ups-account-number>`
+linked to the UPS account created in the previous step.
+
+.. image:: ups_credentials/developer-account-setup.png
+   :align: center
+   :alt: Show form to fill in the UPS account number.
+
+Click :guilabel:`Next`, and proceed to the :guilabel:`Add App` form, and fill out the fields:
+
+- :guilabel:`App Name`: Type the name to identify the app by.
+- :guilabel:`Callback URL`: Type the URL of the Odoo database, in the format: `https://` + database
+  name + `.odoo.com`. Do **not** include `www.`
+
+In the :guilabel:`Add Products` section on the right, search for and click the :guilabel:`+ (plus)`
+icon to add the following products to the app:
+
+- :guilabel:`Authorization (O Auth)`: Used to generate the authorization token to request
+  information from the UPS API.
+- :guilabel:`Address Validation`: Validates addresses at the street level in the United States and
+  Puerto Rico.
+- :guilabel:`Locator`: Enables search for UPS shipping locations based on type and available
+  services.
+- :guilabel:`Paperless Documents`: Enables the upload of document images to link to shipments.
+- :guilabel:`Shipping`: Enables UPS shipping services, such as preparing packages for shipment,
+  managing returns, and canceling scheduled shipments.
+- :guilabel:`Rating`: Compare delivery services and shipping rates.
+
+.. seealso::
+   `UPS API Catalog <https://developer.ups.com/catalog?loc=en_US>`_
+
+.. image:: ups_credentials/add-app-development.png
+   :align: center
+   :alt: Show "Add Apps" form, where the app details are configured.
+
+Finally, click :guilabel:`Save` and accept UPS's terms and conditons. With the new app created,
+select the app from the :guilabel:`My Apps` section to view the UPS credentials.
+
+.. image:: ups_credentials/my-apps.png
+   :align: center
+   :alt: Show newly created app in the "My Apps" section.
+
+In the :guilabel:`Credentials` section, copy the :guilabel:`Client ID` and :guilabel:`Client Secret`
+key.
+
+.. _inventory/shipping_receiving/ups-client-id:
+
+Client ID
+---------
+
+To obtain the account's :guilabel:`Client ID`, click the profile and select :menuselection:`Apps`
+from the drop-down menu, and select the Odoo shipping account.
+
+In the :guilabel:`Credentials` section, copy the :guilabel:`Client ID`.
+
+.. image:: ups_credentials/credentials.png
+   :align: center
+   :alt: Display the "Client ID" and "Client Secret" key.
+
+.. _inventory/shipping_receiving/ups-client-secret:
+
+Client Secret
+-------------
+
+To obtain the account's :guilabel:`Client Secret` key, click the profile and select
+:menuselection:`Apps` from the drop-down menu, and select the Odoo shipping account.
+
+In the :guilabel:`Credentials` section, copy the :guilabel:`Client Secret`.
```

Comment:
The instructions here seem redundant and actually aren't as clear or concise as line 138, while we're in the **My Apps > App > Credentials** section.

I think you can remove this whole block, but keep the screenshot since that's relevant to the instruction on line 138.